### PR TITLE
Move to new `io`

### PR DESCRIPTION
**Pull request by hauleth**
# WIP - DO NOT MERGE

Hell lot of compile errors. It is just for TravisCI reports.

### CODE REVIEW

**Comment by nwin** [comment]
I would wait with that. The docs still recommend using `old_io`. We also have to careful review this, because the semantics of `io` change. For example EOF is not an error any more and I don’t remember whether we exploited that somewhere.

File: src/utils/bitstream.rs [hunk 1]
```diff
@@ -1,72 +1,75 @@
 //! This module provides a bit reader
 
-use std::old_io;
-use std::old_io::{IoResult};
+use std::io::{
+    self,
```
Comment (nwin) [inline src/utils/bitstream.rs]: Is this new? I didn’t know you could do that.

**Comment by hauleth** [comment]
It is preparing for change. I know that this take time to stabilize new `std::io`. But this will be hell lot of work to change it afterwards. We can do some of it aheado of time.

**Comment by nwin** [comment]
I see your point. I just don’t want to force the people depending on `image` to use `io` as long as the docs recommend against using it. It’s good that you prepare everything but I don’t think we should merge before they remove that warning.

**Comment by hauleth** [comment]
That's why there is big `WIP - DO NOT MERGE` at the top. I will make it bigger.

File: src/utils/lzw.rs [hunk 2]
```diff
@@ -268,8 +270,8 @@ impl EncodingDict {
     }
 }
 
-pub fn encode<R, W>(mut r: R, mut w: W, min_code_size: u8) -> old_io::IoResult<()>
-where R: Reader, W: BitWriter {
+pub fn encode<R, W>(mut r: R, mut w: W, min_code_size: u8) -> io::Result<()>
```
Comment (nwin) [inline src/utils/lzw.rs]: This function has to be rewritten. `while let Ok(c) = ...` will be an infinite loop with `std::io`;

**Comment by bvssvni** [comment]
@hauleth Thanks for working on this!

**Comment by nwin** [comment]
Absolutely! 

One other remark it is probably the easiest to copy the read_le_u16() methods from old_io into a new extention trait and implement it for all `Read` like we did for `Primitive`. That will save a lot of pain.

**Comment by hauleth** [comment]
@nwin, there is no need in creating that traits. There already is [podio](https://github.com/mvdnes/podio). We can use this.

**Comment by jpernst** [comment]
Looks like the warning to continue using `old_io` has been removed, so this should probably be high priority. I imagine lots of people are going to take a look at rust during beta 1, particularly for game dev, so we should try to have this ready for the influx. I took a stab at doing this conversion a few weeks ago but it's such a mammoth change for a codebase I don't know very well that I quickly got buried in the avalanche of errors and just swapped it out for a C library binding. I'd prefer the all rust solution, but I'm not sure how to make the changes incrementally without the whole thing coming apart.

**Comment by bvssvni** [comment]
I opened https://github.com/PistonDevelopers/image/issues/333

File: src/utils/bitstream.rs (hunk 1 above)
Comment (bombless) [inline src/utils/bitstream.rs]: @nwin It was added months ago.
See https://github.com/rust-lang/rust/pull/20365

**Comment by bvssvni** [comment]
Fixed by https://github.com/PistonDevelopers/image/pull/339

**Pull request closed without merging**